mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+        ------
+        ValueError
+            If delta is ``None`` and method is RCSP
+        ValueError
+            If delta is not in [0, 1] and method
+            is RCPS.
+        """
+        if (self.method == "rcps") and (delta is None):
+            raise ValueError(
+                "Invalid delta. "
+                "delta cannot be None when using "
+                "RCPS method."
+            )
+        if (not isinstance(delta, float)) and (delta is not None):
+            raise ValueError(
+                "Invalid delta. "
+                f"delta must be a float, not a {type(delta)}"
+            )
+        if (delta is not None) and (
+            ((delta <= 0) or (delta >= 1)) and
+            (self.method == "rcps")
+        ):
+            raise ValueError(
+                "Invalid delta. "
+                "delta must be in ]0, 1["
+            )
+
+    def _check_estimator(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        estimator: Optional[ClassifierMixin] = None,
+    ) -> ClassifierMixin:
+        """
+        Check if estimator is ``None``,
+        and returns a multi-output ``LogisticRegression``
+        instance if necessary.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+
+        estimator : Optional[ClassifierMixin], optional
+            Estimator to check, by default ``None``
+
+        Returns
+        -------
+        ClassifierMixin
+            The estimator itself or a default multi-output
+            ``LogisticRegression`` instance.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not ``None``
+            and has no fit, predict, nor predict_proba methods.
+
+        NotFittedError
+            If the estimator is not fitted.
+        """
+        if estimator is None:
+            return MultiOutputClassifier(
+                LogisticRegression(multi_class="multinomial")
+            ).fit(X, y)
+
+        if isinstance(estimator, Pipeline):
+            est = estimator[-1]
+        else:
+            est = estimator
+        if (
+            not hasattr(est, "fit")
+            or not hasattr(est, "predict")
+            or not hasattr(est, "predict_proba")
+        ):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a classifier with fit,"
+                "predict, and predict_proba methods."
+            )
+        check_is_fitted(est)
+        return estimator
+
+    def _check_partial_fit_first_call(self) -> bool:
+        """
+        Check that this is the first time partial_fit
+        or fit is called.
+
+        Returns
+        -------
+        bool
+            True if it is the first time, else False.
+        """
+        return not hasattr(self, "risks")
+
+    def _check_bound(self, bound: Optional[str]):
```

Comment:
should raise a warning if method is crc and bound is not None
maybe an error?

Reply:
done